src/components/Reports/DonationsReport/DonationsReportTable.tsx

```diff
@@ -1,10 +1,242 @@
+  convertedAmount: number;
+  foreignAmount: number;
+  designation: string;
+  method: string;
+  id: string;
+}
 
 interface Props {
+  data: Donation[];
   accountListId: string;
 }
 
-export const DonationsReportTable: React.FC<Props> = () => {
-  return <Box />;
+const Data = styled(Box)(({ theme }) => ({
+  display: 'flex',
+  flexDirection: 'column',
+  '& .MuiDataGrid-row.Mui-even:not(:hover)': {
+    backgroundColor:
+      theme.palette.type === 'light'
+        ? theme.palette.common.white
+        : theme.palette.cruGrayLight.main,
+  },
+}));
+export const DonationsReportTable: React.FC<Props> = ({
+  data,
+  /*accountListId,*/
+}) => {
+  const { t } = useTranslation();
+  const accountCurrency = data[0].currency;
+  const link = (params: GridCellParams) => (
+    <Typography>
+      <Link href="contact.html">{params.value}</Link>
+    </Typography>
+  );
+  const amount = (params: GridCellParams) => {
+    const row = params.row as Donation;
+
+    return <Typography>{row.convertedAmount + ' ' + row.currency}</Typography>;
+  };
+  const foreignAmount = (params: GridCellParams) => {
+    const row = params.row as Donation;
+    return row.foreignCurrency !== row.currency ? (
+      <Typography>{row.foreignAmount + ' ' + row.foreignCurrency}</Typography>
+    ) : (
+      <></>
+    );
+  };
+  const button = () => (
+    <IconButton color="primary">
+      <EditIcon />
+    </IconButton>
+  );
+  const columns: GridColDef[] = [
+    {
+      field: 'date',
+      headerName: t('Date'),
+      type: 'date',
+      width: 160,
+    },
+    {
+      field: 'partner',
+      headerName: t('Partner'),
+      width: 220,
+      renderCell: link,
+    },
+    {
+      field: 'convertedAmount',
+      headerName: t('Amount'),
+      width: 160,
+      renderCell: amount,
+    },
+    {
+      field: 'foreignAmount',
+      headerName: t('Foreign Amount'),
+      width: 180,
+      renderCell: foreignAmount,
+    },
+    {
+      field: 'designation',
+      headerName: t('Designation'),
+      width: 220,
+    },
+    {
+      field: 'method',
+      headerName: t('Method'),
+      width: 160,
+    },
+    {
+      field: 'appeal',
+      headerName: t('Appeal'),
+      width: 130,
+      renderCell: button,
+    },
+  ];
+  const isEmpty = data.length === 0;
+
+  const [time, setTime] = useState(DateTime.now().startOf('month'));
+
+  const title = time.monthLong + ' ' + time.year;
+
+  const hasNext = time.hasSame(DateTime.now().startOf('month'), 'month');
+
+  const setPrevMonth = () => {
+    setTime(time.minus({ months: 1 }));
+  };
+
+  const setNextMonth = () => {
+    setTime(time.plus({ months: 1 }));
+  };
+
+  const totalDonations = data.reduce((total, current) => {
+    return total + current.convertedAmount;
+  }, 0);
+
+  const calculateForeignTotals = () => {
+    const foreignTotals: {
+      [name: string]: { convertedTotal: number; foreignTotal: number };
+    } = {};
+    data.forEach((donation) => {
```

Comment:
I think a `.reduce` could make this cleaner.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce